### PR TITLE
Newsletters: add 235 (2023-01-25)

### PR DESCRIPTION
**Pull request by harding**
- [ ] Releases/RCs and topic links @harding 
- [x] Bitcoin Stack Exchange @bitschmidty 
- [x] Bitcoin Core 26325 @Xekyo

### CODE REVIEW

**Review by Xekyo — COMMENTED** [review]
LGTM

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  more inputs are used with presigned transactions.  The presigned
+  nature of these transactions implies that fees may need to be added
+  later when an appropriate feerate is known, and the existing
+  `SIGHASH_ANYONECANPAY` and `SIGHASH_SINGLE` sighash flags aren't
+  flexible enough for multi-input transactions because they only commit
+  to a single input or output.
+
+    Ephemeral anchors, similar to [fee sponsorship][topic fee
+    sponsorship], lets anyone [CPFP][topic cpfp] fee bump a transaction.
+    The transaction being fee bumped is allowed to contain zero fees.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:37]:
```suggestion
    The transaction being fee-bumped is allowed to contain zero fees.
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    Because anyone can fee bump a transaction using ephemeral anchors,
+    this mechanism can also be used to pay fees for the multi-input
+    presigned transactions which are a target for `SIGHASH_GROUP`.
+
+    `SIGHASH_GROUP` would still have two advantages: first, that it
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:42]:
```suggestion
    `SIGHASH_GROUP` would still have two advantages: first, it
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    could allow [batching][topic payment batching] multiple unrelated
+    presigned transactions, which could reduce transaction size
+    overhead, which would reduce user costs and increase network
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:45]:
```suggestion
    overhead, reducing user costs and increasing network
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    capacity.  Second, that it doesn't require a child transaction,
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:46]:
```suggestion
    capacity.  Second, it doesn't require a child transaction,
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    which would further reduce costs and increase capacity.
+
+    Towns concludes by noting that ephemeral anchors, with its
+    dependency of [v3 transaction relay][topic v3 transaction relay],
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:50]:
```suggestion
    dependency on [v3 transaction relay][topic v3 transaction relay],
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    capture most of the benefits of `SIGHASH_GROUP` and provides the
```
[the remark this answers is not in the document]
Comment (LarryRuane) [reply _posts/en/newsletters/2023-01-25-newsletter.md:51]: I'd say your first suggestion is correct since the subject is singular -- "ephemeral anchors", even though it ends in "s", is singular because it's a concept. I considered suggesting "the ephemeral anchors concept" (or "approach") on line 49 but I think that's too wordy.

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #26325][] improves the results of the `scanblocks` rpc
+  by removing false positives in a second pass. `scanblock` may be used
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:103]:
```suggestion
  by removing false positives in a second pass. `scanblocks` may be used
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  to find blocks transactions relevant to a provided set of descriptors.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:104]:
```suggestion
  to find blocks containing transactions relevant to a provided set of descriptors.
```

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  Since scanning against the filters may falsely indicate some blocks
+  that do not actually contain relevant transactions, this PR validates
+  each hit in another pass to see whether blocks actually correspond to
+  the passed descriptors before providing the results to the caller.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-01-25-newsletter.md:108]: It may be worth mentioning that this functionality is disabled by default (for performance reasons); the user must specify `-filter_false_positives` to enable it.

**Review by LarryRuane — COMMENTED** [review]
I verified the SE question numbers, and lightly reviewed the content of the summaries.

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Bitcoin Core signing keys were removed from repo. What is the new process?]({{bse}}116649)
+  Andrew Chow explains that while the signing keys were [removed][remove builder
+  keys] from the Bitcoin Core repository, there is now a list of keys on the
+  [guix.sigs repository][guix.sigs repo] which houses [guix][topic reproducible
+  builds] build attestations.
+
+- [Why doesn't signet use a unique bech32 prefix?]({{bse}}116630)
+  Casey Rodarmor wonders why both testnet and [signet][topic signet] both use
```
[the remark this answers is not in the document]
Comment (bitschmidty) [reply _posts/en/newsletters/2023-01-25-newsletter.md]: Thanks Larry, took this suggestion

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+  the `tb1` [address prefix][wiki address prefixes]. Kalle, one of the [BIP325][] authors, explains that
+  while signet initially used a different address prefix, it was thought using the same
+  prefix would simplify usage of that alternate test network.
+
+- [Arbitrary data storage in witness?]({{bse}}116875)
+  RedGrittyBrick points to [one of several][large witness tx] recent P2TR
+  transactions containing a large amount of witness data. Other users point out
+  that the Ordinals project provides a service for including arbitrary data,
+  like [the image][ordinals example] in the transaction above, into a Bitcoin transaction using the witness.
```
[the remark this answers is not in the document]
Comment (bitschmidty) [reply _posts/en/newsletters/2023-01-25-newsletter.md:99]: I didn’t think the response was notable (as far as sipa answers go anyways!), personally. Other than noting it was a P2TR, which I did add for context. Which part do you suggest?

[the remark this answers is not in the document]
Comment (LarryRuane) [reply _posts/en/newsletters/2023-01-25-newsletter.md:99]: Okay; I thought you might not be aware that he had answered, since it was so recent.

I thought it was interesting (and new to me) that P2TR spends' standardness rules are "much more relaxed" than (at least) segwit v0 rules. But now that I read his answer again, he is not sure if the transactions the question is about are considered standard or not. So you're right, not worth mentioning here. 

I also thought that the summaries here _always_ include at least something from the answers given, not just the question. But there's no reason we can't just highlight a question and let readers follow the link if they want to know more. So, in short, never mind, it's okay as is.

File: _posts/en/newsletters/2023-01-25-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+  allows an input to specify which outputs it authorizes, with
+  different inputs in a transaction being able to specify different
+  outputs as long as they don't overlap.  This can be especially useful
+  for adding fees to transactions in contract protocols where two or
+  more inputs are used with presigned transactions.  The presigned
+  nature of these transactions implies that fees may need to be added
+  later when an appropriate feerate is known, and the existing
+  `SIGHASH_ANYONECANPAY` and `SIGHASH_SINGLE` sighash flags aren't
+  flexible enough for multi-input transactions because they only commit
+  to a single input or output.
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2023-01-25-newsletter.md:33]: I think or is more correct, as `SIGHASH_SINGLE` signs multiple outputs (unless a transaction only has one output).

[the remark this answers is not in the document]
Comment (LarryRuane) [reply _posts/en/newsletters/2023-01-25-newsletter.md:33]: Are you sure? Antonop's book (table 4 of the section I linked to above) says
> SINGLE - Signature applies to all inputs but only the one output with the same index number as the signed input
> SINGLE|ANYONECANPAY - Signature applies to one input and the output with the same index number

File: _posts/en/newsletters/2023-01-25-newsletter.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (bitschmidty) [reply _posts/en/newsletters/2023-01-25-newsletter.md:99]: Yeah, there isnt a great answer yet on that one but I thought it was interesting.

Also found per their docs: 

> Inscription content is included in transaction witnesses, which receive the witness discount. To calculate the approximate fee that in inscribe transaction will pay, divide the content size by four and muliply by the fee rate.

> Inscription transactions must be less than 400,000 weight units, or they will not be relayed by Bitcoin Core. One byte of inscription content costs one weight unit. Since an inscription transaction includes not just the inscription content, limit inscription content to less than 400,000 weight units. 390,000 weight units should be safe.

https://docs.ordinals.com/guides/inscriptions.html

So it sounds like it is relayed under 400,000wu

File: _posts/en/newsletters/2023-01-25-newsletter.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2023-01-25-newsletter.md:33]: Sorry, I meant it signs multiple inputs.

**Comment by harding** [comment]
Made edits for all feedback (thanks @glozow @Xekyo @LarryRuane !).  Added a summary for libsecp256k1 1192 with the help of @apoelstra and @sipa (thanks!), added a topic page for ephemeral anchors, reviewed @bitschmidty's section (thanks!), removed the releases/RC section (none this week), and added the topic links.

Thanks everyone!

**Comment by bitschmidty** [comment]
Squashed and merged! Thank you @harding @Xekyo @sipa @apoelstra for authoring and @LarryRuane @glozow @Xekyo for reviewing! 🚀
